### PR TITLE
Development workflow documentation for the current state of the world.

### PR DESCRIPTION
**Pull request by mccheah**
Note the modules section and the integration tests section isn't too much of a concern since we're changing those pretty soon anyways, at least ideally. The current state of how to run integration tests isn't great and we should be looking into improving that.

### CODE REVIEW

**Comment by mccheah** [comment]
@ash211 @foxish @iyanuobidele

File: resource-managers/kubernetes/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# Running the Kubernetes Integration Tests
+
+Note that the integration test framework is currently being heavily revised and is subject to change.
+
+Running any of the integration tests requires including `kubernetes-integration-tests` profile in the build command. In
+order to prepare the environment for running the integration tests, the `pre-integration-test` step must be run in Maven
+on the `resource-managers/kubernetes/integration-tests` module:
+
+  build/mvn pre-integration-test -Pkubernetes -Pkubernetes-integration-tests -pl resource-managers/kubernetes/integration-tests -am
```
Comment (ash211) [inline resource-managers/kubernetes/README.md:48]: once this pre-integration-test profile is run, how do I run the tests themselves?  just the k8s ones

File: resource-managers/kubernetes/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+layout: global
+title: Spark on Kubernetes Development
```
Comment (ash211) [inline resource-managers/kubernetes/README.md:3]: at the bottom of the page we should link to the user docs for how to use it as a natural continuation from the dev setup docs

**Review by ash211 — COMMENTED** [review]
The main things I'd want when looking at dev docs are 1) how to build it, and 2) how to run the tests.  From the user docs, I'd expect to get 3) how to run it.

I think we cover 1 and 3 but not fully 2 yet

File: resource-managers/kubernetes/README.md (hunk 1 above)
Comment (mccheah) [reply resource-managers/kubernetes/README.md:48]: This depends so I wasn't sure how to include it. If just running from the command line, there should be just a maven command for it - this follows the general maven semantics but needing to specify each of the suites means that if we add, rename, or delete suites, we would have to adjust the docs accordingly.

If running in IntelliJ, before running any test in IntelliJ, if any code changes the pre-integration-test has to be run. From there it's just using JUnit to run the test classes.

Comment (ash211) [reply resource-managers/kubernetes/README.md:48]: Maybe then just a message afterwards saying something like "now run your tests normally, either on the command line with maven or through an IDE like IntelliJ"

File: resource-managers/kubernetes/README.md (hunk 2 above)
Comment (iyanuobidele) [reply resource-managers/kubernetes/README.md:3]: +1

File: resource-managers/kubernetes/README.md (hunk 1 above)
Comment (mccheah) [reply resource-managers/kubernetes/README.md:48]: Done

File: resource-managers/kubernetes/README.md (hunk 2 above)
Comment (mccheah) [reply resource-managers/kubernetes/README.md:3]: Done

File: resource-managers/kubernetes/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Note that the integration test framework is currently being heavily revised and is subject to change.
+
+Running any of the integration tests requires including `kubernetes-integration-tests` profile in the build command. In
+order to prepare the environment for running the integration tests, the `pre-integration-test` step must be run in Maven
+on the `resource-managers/kubernetes/integration-tests` module:
+
+  build/mvn pre-integration-test -Pkubernetes -Pkubernetes-integration-tests -pl resource-managers/kubernetes/integration-tests -am
+ 
+Afterwards, the integration tests can be executed with Maven or your IDE. Note that when running tests from an IDE, the
+`pre-integration-test` phase must be run every time the core Kubernetes code changes. When running tests from the
```
Comment (ash211) [inline resource-managers/kubernetes/README.md]: by "core Kubernetes code" here you mean the spark-kubernetes integration, not the upstream kubernetes.io code right?

**Review by ash211 — APPROVED** [review]
Good pending a clarification around "core kubernetes code"

**Comment by foxish** [comment]
Do we also want to link the proposal, https://github.com/kubernetes/kubernetes/issues/34377 and the JIRA here?

File: resource-managers/kubernetes/README.md (hunk 3 above)
Comment (mccheah) [reply resource-managers/kubernetes/README.md]: We should say Spark code in general since people can make changes to core as well.

**Comment by mccheah** [comment]
@foxish good idea but I think just the JIRA will suffice since the proposal is attached to it.

**Comment by foxish** [comment]
Sounds good. Rest LGTM.

**Comment by mccheah** [comment]
@aash addressed clarification, good to merge?

**Pull request merged**